Sources/AblyChat/Messages.swift

```diff
@@ -54,21 +54,43 @@ public struct QueryOptionsWithoutDirection: Sendable {
 public struct MessageSubscription: Sendable, AsyncSequence {
     public typealias Element = Message
 
-    public init<T: AsyncSequence>(mockAsyncSequence _: T) where T.Element == Element {
-        fatalError("Not yet implemented")
+    private var subscription: Subscription<Element>
+
+    private typealias MockGetPreviousMessages = @Sendable (QueryOptionsWithoutDirection) async throws -> any PaginatedResult<Message>
+    private var mockGetPreviousMessages: MockGetPreviousMessages?
+
+    internal init(bufferingPolicy: BufferingPolicy) {
+        subscription = .init(bufferingPolicy: bufferingPolicy)
+    }
+
+    public init<T: AsyncSequence & Sendable>(mockAsyncSequence: T, mockGetPreviousMessages _: (QueryOptionsWithoutDirection) async throws -> any PaginatedResult<Message>) where T.Element == Element {
```

Comment:
Also why not use `QueryOptions` ignoring direction?

Reply:
I think that's a question for https://github.com/ably-labs/ably-chat-swift/pull/9, which defines the signature of `getPreviousMessages`?

Reply:
https://github.com/ably-labs/ably-chat-swift/pull/9/files#r1730072974